### PR TITLE
Fix React, add docs

### PR DESCRIPTION
**Pull request by kossnocorp**
- An incorrect method was used for React version of the library (it was released with `v0.5.1`).
- Rewrite README and add JSDoc entry in the source code.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,81 +1,172 @@
...
+```bash
+npm install decss-loader --save-dev
+# or
+yarn add --dev decss-loader
 ```
 
-React:
-``` js
-import { Loader } from './style.css'
+To use low-level API, or to manage the core library version, install `decss`:
```
Comment (leshakoss) [inline README.md]: **the** low-level API

File: README.md [hunk 2]
```diff
@@ -1,81 +1,172 @@
...
-### Enum properties
+## Convention
 
-CSS:
-``` css
-.Loader-color-green {
-  background: green;
+### Component (`.Component`)
+
+`.Component` is a component 🤡 The name must be in class-case, e.g. `.FormInput`, `.UI`, etc:
```
Comment (leshakoss) [inline README.md]: e.g. `.FormInput`, `.UI`. ~~, etc:~~

File: index.js [hunk 3]
```diff
@@ -1,5 +1,46 @@
 module.exports = decss
 
+/**
+ * @name decss
+ * @summary It generates components using passed CSS modules object.
```
Comment (leshakoss) [inline index.js]: I think this should be in imperative (`Generate a component using passed CSS modules object.`) but I may be wrong here

File: README.md [hunk 4]
```diff
@@ -1,81 +1,172 @@
 # decss
 
-CSS components
+decss converts [CSS modules] to React/Preact components. Thanks to
```
Comment (leshakoss) [inline README.md]: I would highlight `decss` here and everywhere else with bold or italic font

File: README.md [hunk 5]
```diff
@@ -1,81 +1,172 @@
...
+a simple CSS class naming convention it allows to automatically
+generate enum and boolean props so that you can keep your code
+clean and tidy.
```
Comment (leshakoss) [inline README.md]: Whatever, it was just a weird sentence. I would write like this:

```markdown
**decss** converts [CSS modules] into React/Preact components. Thanks to
simple CSS class naming convention it
generates enum and boolean props for the component automatically
so you can keep your code clean and tidy.
```

File: README.md [hunk 6]
```diff
@@ -1,81 +1,172 @@
...
+`.*-disabled` is a boolean prop.
+
+```css
+.Button-disabled {
+  opacity: .5;
 }
 ```
 
-### Default properties
+In the example, `.Button-disabled` is applied to `.Button`
```
Comment (leshakoss) [inline README.md]: In the example, `.Button-disabled` is applied to ~~`.Button`~~ the component
when **its** `disabled` prop is truthy:

File: README.md [hunk 7]
```diff
@@ -1,81 +1,172 @@
...
+
+.Button-color-green {
   background: green;
 }
 ```
 
-React (Loder is green by default):
-``` js
-import { Loader } from './style.css'
+`.Button-color-red` is applied to `.Button` when `color` prop equals `"red"`:
```
Comment (leshakoss) [inline README.md]: `.Button-color-red` is applied to the component when its `color` prop equals `"red"`:

**Comment by kossnocorp** [comment]
@leshakoss thanks!

File: README.md (hunk 4 above)
Comment (kossnocorp) [inline README.md]: @leshakoss why? I think that's fine.

Comment (leshakoss) [inline README.md]: Well, you did highlight **webpack** in 12th line for some reason

Comment (kossnocorp) [inline README.md]: I wanted to highlight an important term that people could search for. I'll make it consistent.

**Pull request merged**